group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2032,11 +2031,10 @@ private CoordinatorResult<ShareGroupHeartbeatResponseData, CoordinatorRecord> sh
 
         // Get or create the share group.
         boolean createIfNotExists = memberEpoch == 0;
-        final ShareGroup group = getOrMaybeCreatePersistedShareGroup(groupId, createIfNotExists);
+        final ShareGroup group = getOrMaybeCreateShareGroup(groupId, createIfNotExists);
```

Comment:
Why do we want to change this behaviour?

Reply:
I would ask this the other way around. Why is this different than consumer groups? I have simply aligned it with consumer groups, including the structure of the if-else blocks.

Reply:
I have to recall, there might be reasons to move differently or a mistake. I ll look into the details again.

Reply:
Seems like it was a mistake and all test cases pass 👍.